python/flatbuffers/builder.py

```diff
@@ -238,11 +245,11 @@ def EndObject(self):
     def growByteBuffer(self):
         """Doubles the size of the byteslice, and copies the old data towards
            the end of the new buffer (since we build the buffer backwards)."""
-        if not len(self.Bytes) <= 2**20:
+        if not len(self.Bytes) <= Builder.MAX_BUFFER_SIZE:
             msg = "flatbuffers: cannot grow buffer beyond 2 gigabytes"
             raise BuilderSizeError(msg)
 
-        newSize = len(self.Bytes) * 2
+        newSize = min( len(self.Bytes) * 2, Builder.MAX_BUFFER_SIZE )
```

Comment:
This change and the one above lead to the incorrect behavior: we can call `growByteBuffer` as many times as we want and it will never raise `BuilderSizeError` because `newSize` can't become more than `Builder.MAX_BUFFER_SIZE`.
I suggest to fix it that way:
```python
if len(self.Bytes) == Builder.MAX_BUFFER_SIZE:
    msg = "flatbuffers: cannot grow buffer beyond 2 gigabytes"
    raise BuilderSizeError(msg)
 
newSize = min(len(self.Bytes) * 2, Builder.MAX_BUFFER_SIZE)
```
It ensures that the last growing will make buffer as big as possible and that succeeding calls to `growByteBuffer` will raise an exception.